src/test/java/com/verisign/storm/metrics/reporters/BaseKafkaReporterTest.java

```diff
@@ -0,0 +1,288 @@
+import com.verisign.storm.metrics.reporters.kafka.BaseKafkaReporter;
+import com.verisign.storm.metrics.serializers.AvroRecordSerializer;
+import io.confluent.kafka.schemaregistry.client.LocalSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import io.confluent.kafka.serializers.KafkaAvroDecoder;
+import io.confluent.kafka.serializers.KafkaAvroSerializer;
+import kafka.api.FetchRequest;
+import kafka.api.FetchRequestBuilder;
+import kafka.javaapi.FetchResponse;
+import kafka.javaapi.consumer.SimpleConsumer;
+import kafka.message.MessageAndOffset;
+import kafka.server.KafkaConfig;
+import kafka.server.KafkaServerStartable;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.Decoder;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.avro.specific.SpecificRecordBase;
+import org.apache.curator.test.TestingServer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.common.errors.SerializationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.util.*;
+
+import static org.fest.assertions.api.Assertions.assertThat;
+import static org.fest.assertions.api.Assertions.fail;
+
+public class BaseKafkaReporterTest {
+  private int testCount = 10;
+  private Logger LOG = LoggerFactory.getLogger(BaseKafkaReporterTest.class);
+
+  private String testZkHost = "127.0.0.1";
+  private Integer testZkPort = 2181;
+  private String testZkConnect = testZkHost + ":" + testZkPort;
+
+  private String testKafkaHost = "127.0.0.1";
+  private Integer testKafkaPort = 9092;
+  private String testKafkaBrokerList = testKafkaHost + ":" + testKafkaPort;
+
+  private String destinationTopic;
+  private KafkaServerStartable kafkaServer;
+  private TestingServer zookeeper;
+  private BaseKafkaReporter kafkaReporter;
+  private LocalSchemaRegistryClient schemaRegistryClient;
+
+  @BeforeClass private void initializeCluster() {
+    try {
+      zookeeper = new TestingServer(testZkPort);
+    }
+    catch (Exception e) {
+      LOG.error(e.getMessage());
+    }
+
+    KafkaConfig kafkaConfig = new KafkaConfig(getBrokerConfig());
+    kafkaServer = new KafkaServerStartable(kafkaConfig);
+    kafkaServer.startup();
+  }
+
+  @AfterClass private void exitCluster() {
+    kafkaServer.shutdown();
+    try {
+      zookeeper.close();
+    }
+    catch (IOException e) {
+      LOG.error("Failed to close Zookeeper server at {}:{}", testZkHost, testZkPort);
+    }
+  }
+
+  @DataProvider(name = "metrics") public Object[][] metricsProvider() {
```

Comment:
The data provider feeds into both tests (for the `AvroKafkaReporter` and `SchemaRegistryKafkaReporter`).